src/data_types.js

```diff
@@ -229,14 +229,24 @@ class DATE extends DataType {
   toSqlString() {
     const { precision } = this;
     const dataType = this.dataType.toUpperCase();
-    if (precision > 0) return `${dataType}(${precision})`;
+    if (precision != null && precision >= 0) return `${dataType}(${precision})`;
     return dataType;
   }
 
+  _round(value) {
+    const { precision } = this;
+    if (precision != null && precision < 3 && value instanceof Date) {
+      const result = new Date(value);
+      result.setMilliseconds(result.getMilliseconds() % (10 ** precision));
```

Comment:
`result.getMilliseconds() % (10 ** precision)` seems not correct?
while `precision  = 2, Milliseconds = 864`, expect `86`
```js
864 % (10 ** 2) = 64
```
how about `Number(result.getMilliseconds()/ (10 ** (3- precision)).toFixed())`

Reply:
nice catch! updated with following solution:

```js
const divider = 10 ** (3 - precision);
 return new Date(Math.round(value.getTime() / divider) * divider);
```